vendor/bat-native-ledger/src/bat/ledger/internal/report/report.cc

```diff
@@ -0,0 +1,168 @@
+
+void Report::GetMonthly(
+      const ledger::ActivityMonth month,
+      const int year,
+      ledger::GetMonthlyReportCallback callback) {
+  auto balance_callback = std::bind(&Report::OnBalance,
+      this,
+      _1,
+      _2,
+      month,
+      year,
+      callback);
+
+  ledger_->GetBalanceReport(month, year, balance_callback);
+}
+
+void Report::OnBalance(
+    const ledger::Result result,
+    ledger::BalanceReportInfoPtr balance_report,
+    const ledger::ActivityMonth month,
+    const uint32_t year,
+    ledger::GetMonthlyReportCallback callback) {
+  if (result != ledger::Result::LEDGER_OK || !balance_report) {
+    BLOG(ledger_, ledger::LogLevel::LOG_ERROR)
+        << "Could not get balance report";
+    callback(result, nullptr);
+    return;
+  }
+
+  auto monthly_report = ledger::MonthlyReportInfo::New();
+  monthly_report->balance = std::move(balance_report);
+
+  const std::string monthly_report_string =
+      braveledger_bind_util::FromMonthlyReportToString(
+          std::move(monthly_report));
+
+  auto transaction_callback = std::bind(&Report::OnTransactions,
+      this,
+      _1,
+      month,
+      year,
+      monthly_report_string,
+      callback);
+
+  ledger_->GetTransactionReport(month, year, transaction_callback);
+}
+
+void Report::OnTransactions(
+    ledger::TransactionReportInfoList transaction_report,
+    const ledger::ActivityMonth month,
+    const uint32_t year,
+    const std::string& monthly_report_string,
+    ledger::GetMonthlyReportCallback callback) {
+
+  auto monthly_report = braveledger_bind_util::FromStringToMonthlyReport(
+      monthly_report_string);
+
+  if (!monthly_report) {
+    BLOG(ledger_, ledger::LogLevel::LOG_ERROR)
+        << "Could not parse monthly report";
+    callback(ledger::Result::LEDGER_ERROR, nullptr);
+    return;
+  }
+
+  monthly_report->transactions = std::move(transaction_report);
+
+  const std::string callback_monthly_string =
+      braveledger_bind_util::FromMonthlyReportToString(
+          std::move(monthly_report));
+
+  auto contribution_callback = std::bind(&Report::OnContributions,
+      this,
+      _1,
+      callback_monthly_string,
+      callback);
+
+  ledger_->GetContributionReport(month, year, contribution_callback);
+}
+
+void Report::OnContributions(
+    ledger::ContributionReportInfoList contribution_report,
+    const std::string& monthly_report_string,
+    ledger::GetMonthlyReportCallback callback) {
+
+  auto monthly_report = braveledger_bind_util::FromStringToMonthlyReport(
+      monthly_report_string);
+
+  if (!monthly_report) {
+    BLOG(ledger_, ledger::LogLevel::LOG_ERROR)
+        << "Could not parse monthly report";
+    callback(ledger::Result::LEDGER_ERROR, nullptr);
+    return;
+  }
+
+  monthly_report->contributions = std::move(contribution_report);
+
+  callback(ledger::Result::LEDGER_OK, std::move(monthly_report));
+}
+
+// This will be removed when we move reports in database and just order in db
```

Comment:
Is our convention to use `TODO` for these kinds of notes?

Reply:
it's not really a TODO, it's more a note